src/frontend/src/table.rs

```diff
@@ -348,6 +352,26 @@ impl DistTable {
         };
         Ok(partition_rule)
     }
+
+    /// Define a `alter_by_expr` instead of impl [`Table::alter`] to avoid redundant conversion between  
+    /// [`table::requests::AlterTableRequest`] and [`AlterExpr`].
+    pub(crate) async fn alter_by_expr(&self, expr: AlterExpr) -> Result<()> {
+        let table_routes = self.table_routes.get_route(&self.table_name).await.unwrap();
+        for datanode in table_routes.find_leaders() {
```

Comment:
If "leaders" are empty, I think we'd better return an error. After all, the "alter" is not actually applied.